onecodex/models/sample.py

```diff
@@ -75,6 +80,23 @@ def where(cls, *filters, **keyword_filters):
             # TODO: implement this (see above block)
             samples = metadata_samples
 
+        # finally, filter by tags
+        if tag:
+            samples_to_keep = []
+
+            for s in samples:
+                try:
+                    if tag in s.tags:
+                        samples_to_keep.append(s)
+                except HTTPError:
+                    # this is here for some strange error where a sample has a tag that doesn't exist?
+                    pass
+
+                if len(samples_to_keep) == limit:
+                    break
+
+            samples = samples_to_keep
```

Comment:
See comment above re: removing and simplifying this logic (plus shipping less data!).